releasetool/commands/tag/java.py

```diff
@@ -0,0 +1,133 @@
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import getpass
+import re
+from typing import Optional
+
+import attr
+import click
+
+import releasetool.circleci
+import releasetool.git
+import releasetool.github
+import releasetool.secrets
+import releasetool.commands.common
+
+
+@attr.s(auto_attribs=True, slots=True)
+class Context(releasetool.commands.common.GitHubContext):
+    package_name: Optional[str] = None
+    release_pr: Optional[dict] = None
+    release_tag: Optional[str] = None
+    release_version: Optional[str] = None
+    release_notes: Optional[str] = None
+    github_release: Optional[dict] = None
+
+
+def determine_release_pr(ctx: Context) -> None:
+    click.secho(
+        "> Let's figure out which pull request corresponds to your release.", fg="cyan"
+    )
+
+    pulls = ctx.github.list_pull_requests(ctx.upstream_repo, state="closed")
+    pulls = [pull for pull in pulls if "release" in pull["title"].lower()][:30]
+
+    click.secho("> Please pick one of the following PRs:\n")
+    for n, pull in enumerate(pulls, 1):
+        print(f"\t{n}: {pull['title']} ({pull['number']})")
+
+    pull_idx = click.prompt("\nWhich one do you want to tag and release?", type=int)
+
+    ctx.release_pr = pulls[pull_idx - 1]
+
+
+def determine_release_tag(ctx: Context) -> None:
+    click.secho("> Determining what the release tag should be.", fg="cyan")
+    head_ref = ctx.release_pr["head"]["ref"]
+    match = re.match("release-.+-(v\d+\.\d+\.\d+)", head_ref)
+
+    if match is not None:
+        ctx.release_tag = match.group(1)
+    else:
+        print(
```

Comment:
`click.secho(..., fg="red")`? (or yellow)

Reply:
Copied from the nodejs tag command - updated both.